### PR TITLE
Planning Scene transcription V2

### PR DESCRIPTION
**Pull request by kamiradi**
@sea-bass @sjahr. This is at a reviewable stage. The planning scene transcription contains the following
- Transcribes Box, Sphere, Cylinder primitives directly from the planning scene object (tested on box, yet to test on sphere and cylinder)
- I tried going down the octomap road, but I could not see an obvious way of transcribing it in drake. Any thoughts? 

I'll do the pre-commit stuff after any other changes you guys have in mind

### CODE REVIEW

File: .docker/Dockerfile
Comment (sea-bass) [inline .docker/Dockerfile:41]: What was this needed for?

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:259]: Remove before merging

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:264]: Remove before merging, or make into debug? Might still be useful, long as it doesn't spam by default

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:270]: Same as other comment. In general, should clean up what to print. So I'll stop commenting the same thing

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:271]: The name should probably not be "template_SHAPE" but rather the actual entity name from the planning scene?

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:289]: Maybe this should be a warning instead

**Review by sea-bass — COMMENTED** [review]
Changes look good!

I think all I see is cleaning up some of the debug prints and the hard-coded names -- ensure the planning scene object names match those specified here? But maybe I am misunderstanding how the Drake scene graph works.

File: .docker/Dockerfile
Comment (sjahr) [inline .docker/Dockerfile:29]: Similar to Sebastian's comment: Why do we need this?

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:234]: Delete?

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:256]: Make this a warning and say explicitly: Octomap parsing not supported at the moment

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:301]: Remove

**Review by sjahr — COMMENTED** [review]
Make sure to run `pre-commit run -a` between the iterations on the code. It makes it also easier to read and review

File: .docker/Dockerfile
Comment (kamiradi) [reply .docker/Dockerfile:29]: I added this for linting and LSP support, so that vim has access to the symbols for functionality like Goto definition/declaration etc.

**Comment by kamiradi** [comment]
@sea-bass @sjahr So I may have celebrated to soon, 
drake_scene_graph_planning_scene

There seems to be some thing I may be missing when it comes to reading the poses of the objects within planning scene into the scene_graph. Is there a transformation between I'm missing between the robot and the world. As far as the scene_graph is concerned, I make the robot spawn at world origin. There could be an initial transformation that I am missing?

**Comment by sea-bass** [comment]
>  Is there a transformation I'm missing between the robot and the world. As far as the scene_graph is concerned, I make the robot spawn at world origin. There could be an initial transformation that I am missing?

You could check the TF widget in RViz to see if there is something amiss there? But as far as world-robot transform is concerned, this should anyways come from the URDF and I would be surprised if this was it.

As another thing to look form maybe the scene objects are expressed with some other transform relative to world? @sjahr would know this better.

**Comment by sea-bass** [comment]
Actually from the screenshot it just seems like no transforms at all are being applied to the boxes in the Drake scene graph. Everything is at the origin with identity transform, far as I can tell?

**Comment by kamiradi** [comment]
I think the poses of individual "boxes" (four in this case) w.r.t the collision object frame of reference may be correct. I think there is some global transformation that is missing here. If that makes sense?

**Comment by sjahr** [comment]
> > Is there a transformation I'm missing between the robot and the world. As far as the scene_graph is concerned, I make the robot spawn at world origin. There could be an initial transformation that I am missing?
> 
> You could check the TF widget in RViz to see if there is something amiss there? But as far as world-robot transform is concerned, this should anyways come from the URDF and I would be surprised if this was it.
> 
> As another thing to look form maybe the scene objects are expressed with some other transform relative to world? @sjahr would know this better.

Not sure about that because it works with the MoveIt planning scene, so I guess all transform should be known. Does the function you added fix the issue @kamiradi ?

**Comment by kamiradi** [comment]
The function I added, changes Moveit/ROS coordinate system to Drake. I thought this would, but it didn't. I'll try printing the positions and debugging.

**Comment by kamiradi** [comment]
All right good news, the pose object coming in from the planning scene itself has pose.x, y, z = 0.0. Can you point me towards where in moveit's planning scene the object markers are being set. Because in RViZ it visualises it correctly, so maybe I can use whichever pose to do the same here?

**Comment by kamiradi** [comment]
drake_scene_plan2
drake_scene_plan2
Ok, solved! There was indeed an object pose that I was missing.

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:252]:
```suggestion
      RCLCPP_WARN(getLogger(), "Octomap not supported for now ... ");
```

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:321]: Are these all possible shape types? If not, you should add a default like this at the end of the switch statement
```
default:
{
  RCLCPP_WARN(getLogger(), "Shape TYPE conversation to drake is not implemented");
}
```

File: src/ktopt_planning_context.cpp
Comment (sjahr) [inline src/ktopt_planning_context.cpp:329]: Remove the whole function since it is not needed?

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:163]: Are you wanting to merge with this?

I had an idea which I could try -- basically have the motion planning terminate but starting the visualization separately in a thread so Meshcat doesn't die instantly, but also doesn't hold up the actual planning?

Not for this PR, but at least the delay should be reduced

File: src/ktopt_planning_context.cpp
Comment (sea-bass) [inline src/ktopt_planning_context.cpp:231]:
```suggestion
  // Transcribe the planning scene into a drake scene graph

**Review by sjahr — APPROVED** [review]
Let's merge once SebC's comments are addressed :smile:

**Pull request merged**
